integration/testnet/network.go

```diff
@@ -597,6 +601,23 @@ func BootstrapNetwork(networkConf NetworkConfig, bootstrapDir string) (*flow.Blo
 		if err != nil {
 			return nil, nil, nil, nil, err
 		}
+
+		// We use the network key for the machine account. Normally it would be
+		// a separate key.
+		info := bootstrap.NodeMachineAccountInfo{
+			Address:           nodeConfig.Address,
```

Comment:
Ah, there is a problem here.

These are two different addresses. `nodeConfig.Address` refers to the node's network address, `NodeMachineAccountInfo.Address` refers to the Flow address of the machine account. 

When we create the node machine accounts within the [Register Node transaction](https://github.com/onflow/flow-go/pull/693/files/105c6231575c0bc2791eae219c5493c617a19546..62a75ac86a81a4206530cde9da8ecdff9d5aa521#diff-f414ad868524efa32a2d6a821877f9d01b7948e26bece32df25664a8eb4709d8R724), the node's `Machine Account` is created and assigned an address. Under standard operation, we would retrieve this address by querying `flow.AccountCreated` events for the transaction in which the node was registered. I'm not sure if we have the ability to recover the events within a meta-transaction in the bootstrap context and, even if we do, we will need to pass those events/addresses back across the API boundary of [`BootstrapLedger`](https://github.com/onflow/flow-go/blob/8331b78d99eddea405076e3b6a7839ec5f6ea209/engine/execution/state/bootstrap/bootstrap.go#L31).

I suspect that the simplest way to get these addresses will instead be to leverage the fact that these accounts are generated in a known order during bootstrapping, and account addresses are deterministic based on order for a given chain configuration. During the bootstrapping, it looks like we create accounts for the `FungibleToken`, `FlowToken`, and `FlowFees` smart contracts besides the service account ([code](https://github.com/onflow/flow-go/blob/cae9f3de430230c118c06ba4893ddea979a4c13b/fvm/bootstrap.go#L152-L154)). The service account has index 0, then these 3 accounts would occupy account indices `[1-3]`, so the node machine accounts would occupy account indices `[4,n+4]`. They will be created in the order defined by the [identity list provided to the `BootstrapProcedure`](https://github.com/onflow/flow-go/blob/cae9f3de430230c118c06ba4893ddea979a4c13b/fvm/bootstrap.go#L40).

[`Chain.AddressAtIndex`](https://github.com/onflow/flow-go/blob/5c99b0d8657a613438c9d3a394fbdf555304242c/model/flow/chain.go#L184) can be used to compute the address at each index in `[4, n+4]`.

Reply:
Yeah that's what I feared.
I like the index thing altough it might be a bit confusing to maintain when/if we add account-creating transactions in the bootstrap procedure.
Another option would be to create a global variable in the bootstrap package to track the address of identities.
Actually, the identities are passed by reference to the bootstrap procedure so perhaps we could populate a field directly in the `identities`. It looks like there is already and `address` field there ([code](https://github.com/onflow/flow-go/blob/62665b0ccdbb8f29954d29273d5851530d8ed1d8/model/flow/identity.go#L26)) but I'm not sure if that is supposed to be a network address or a flow address :)
What do you thing?